src/renderer/components/MapEditor/ICCAExportDialog.js

```diff
@@ -0,0 +1,494 @@
+    view and download?`,
+  question3Prompt: `
+    Name of the community, indigenous people, or individual providing the
+    information`,
+  question4Prompt: `
+    If you are an individual, what is your relationship to the indigenous
+    people or community?`,
+  question5Prompt: `
+    Email address, phone number or address where we can contact you. We will
+    contact you to confirm we have received your information and to discuss
+    the information further where needed. We will also contact you periodically
+    to make sure the information is up to date.`,
+  // Yes/no answers
+  answerYes: 'Yes',
+  answerNo: 'No',
+  // Choices for question 4
+  question4Answer1: 'Member of the community/indigenous people',
+  question4Answer2: 'Representative or associate of the community/indigenous people',
+  question4Answer3: 'Representative of a non-governmental organisation',
+  question4Answer4: 'Other',
+  // Placeholder for question 5
+  question5Placeholder: 'Your contact information',
+  // Helper texts
+  requiredAnswer: 'This answer is required.',
+  atLeastOneAnswer: 'Please fill in at least one field.',
+  errorMissingRequired: 'Please fill out all required fields.',
+  // Labels for community name
+  communityOriginalName: 'Original language',
+  communityEnglishName: 'English'
+})
+
+// Insert a tiny lil' shake animation when an error occurs
+// We don't use the makeStyles api here because it can't handle keyframes
+insertCss(`
+  @keyframes shakeX {
+    from,
+    to {
+      transform: translate3d(0, 0, 0);
+    }
+
+    16.67%,
+    50%,
+    83.33% {
+      transform: translate3d(-10px, 0, 0);
+    }
+
+    33.33%,
+    66.67% {
+      transform: translate3d(10px, 0, 0);
+    }
+  }
+
+  .shakeX {
+    animation-name: shakeX;
+    animation-timing-function: ease-in-out;
+    animation-duration: 500ms;
+    animation-delay: 0;
+  }
+`)
+
+const EditDialogContent = ({ onClose }) => {
+  const classes = useStyles()
+  const { formatMessage } = useIntl()
+  const [saving, setSaving] = useState()
+  const [error, setError] = useState(false)
+  const [value1, setValue1] = useState('')
+  const [value2, setValue2] = useState('')
+  const [value3a, setValue3a] = useState('')
+  const [value3b, setValue3b] = useState('')
+  const [value4, setValue4] = useState('')
+  const [value5, setValue5] = useState('')
+  const [value1Error, setValue1Error] = useState(false)
+  const [value2Error, setValue2Error] = useState(false)
+  const [value3Error, setValue3Error] = useState(false)
+  const [value4Error, setValue4Error] = useState(false)
+  const [value5Error, setValue5Error] = useState(false)
+  const ref = useRef()
+
+  // Pre-optimize the parent element for the "shake" animation when there's
+  // an error. If we don't do this, the first time the animation plays,
+  // it feels janky
+  useEffect(() => {
+    if (!ref.current) return
+    ref.current.parentNode.style.willChange = 'transform'
```

Comment:
This approach feels like it is fighting against React and is fragile (not clear what parent node will be if we change things). May I suggest that we add a `onFormError` prop to `EditDialogContent`, and a handler in `ICCAExportDialog` that sets a state `shaking=true`, with a timeout to turn it off, then we add the class `shakeX` to the parent `Dialog` component based on the `shaking` state, using the [`classes` property](https://material-ui.com/api/dialog/) exposed by the Material-ui API.

Reply:
Yep - that's a good call. Thanks for the suggestion.